virtual-display-driver-cli/src/main.rs

```diff
@@ -0,0 +1,343 @@
+use clap::Parser;
+use client::Client;
+use joinery::JoinableIterator;
+use lazy_format::lazy_format;
+use owo_colors::OwoColorize;
+use serde::{Deserialize, Serialize};
+
+mod client;
+
+#[derive(Debug, Parser)]
+struct Args {
+    /// Format output as JSON.
+    #[clap(short, long)]
+    json: bool,
+
+    #[clap(subcommand)]
+    command: Command,
+}
+
+#[derive(Debug, Parser)]
+enum Command {
+    /// List currently connected virtual monitors.
+    List,
+    /// Add a new virtual monitor.
+    Add {
+        /// Width of the virtual monitor.
+        width: driver_ipc::Dimen,
+
+        /// Height of the virtual monitor.
+        height: driver_ipc::Dimen,
+
+        /// More resolutions to add as extra modes for the virtual monitor.
+        more_widths_and_heights: Vec<driver_ipc::Dimen>,
+
+        /// Refresh rate of the virtual monitor. Pass multiple times to
+        /// support multiple refresh rates.
+        #[clap(short, long, default_value = "60")]
+        refresh_rate: Vec<driver_ipc::RefreshRate>,
+
+        /// Manual ID to set for the monitor. Must not conflict with an
+        /// existing virtual monitor's ID.
+        #[clap(long)]
+        id: Option<driver_ipc::Id>,
+
+        /// Optional label for the virtual monitor.
+        #[clap(long)]
+        name: Option<String>,
+
+        /// Set the virtual monitor to disabled on creation.
+        #[clap(long)]
+        disabled: bool,
+    },
+    /// Add a new resolution/refresh rate mode to an existing virtual monitor.
+    AddMode {
```

Comment:
For AddMode, you wrote:
> This adds a mode to an existing virtual monitor. I'm not quite sure if the interface with this one makes perfect sense, since this can lead to duplicate EDIDs and probably other weird things.

This is fine to do. It's pretty much impossible to mess up any internal things in the driver from IPC, no matter what you do. When a mode is added, the following would happen:

1. Monitor updates the internal `Monitor` representation for the one that got changed.
2. It disconnects the monitor.
3. It reconnects the monitor.

Either way, there will never be a duplicate EDID or anything weird happening. 😄

Reply:
_Marking as resolved, this will be handled in a follow-up_